File: autoping.py
import urllib.request
import time
import datetime
import threading
import os

URL = os.environ.get('RENDER_EXTERNAL_URL', 'http://localhost:8000')
INTERVAL = 40  # 40 seconds

def ping():
    while True:
        try:
            response = urllib.request.urlopen(URL)
            status_code = response.getcode()
            print(f"[{datetime.datetime.now().isoformat()}] Pinged {URL} - Status: {status_code}")
        except Exception as e:
            print(f"[{datetime.datetime.now().isoformat()}] Error pinging {URL}: {e}")
        
        time.sleep(INTERVAL)

if __name__ == '__main__':
    print(f"Starting autoping for {URL} every {INTERVAL} seconds...")
    ping()

File: autoping.js
const https = require('https');
const http = require('http');

const url = process.env.RENDER_EXTERNAL_URL || 'http://localhost:8000';
const interval = 5000; // 5 ثواني

console.log(`Starting autoping for ${url} every 5 seconds...`);

setInterval(() => {
  const client = url.startsWith('https') ? https : http;
  client.get(url, (res) => {
    console.log(`[${new Date().toISOString()}] Pinged ${url} - Status: ${res.statusCode}`);
    
    // تفريغ البيانات لتجنب تسرب الذاكرة (Memory Leak)
    res.on('data', () => {});
    res.on('end', () => {});
  }).on('error', (err) => {
    console.error(`[${new Date().toISOString()}] Error pinging ${url}: ${err.message}`);
  });
}, interval);
